lib/mix/lib/mix/sync/lock.ex

```diff
@@ -119,6 +121,12 @@ defmodule Mix.Sync.Lock do
     end
   end
 
+  defp init_base_path(path) do
+    File.mkdir_p!(path)
+    # ensure other users can write to the directory
+    _ = File.chmod(path, 0o777)
```

Comment:
What if this fails? Because another user has created this directory?

Reply:
that's why I ignore the return value. I assume that the user that initially creates the folder has the necessary permissions.

Reply:
Right, but the reason we have a report is exactly because the permissions are confusing. What if we generate a single directory like this: "mix-lock-#{user}-#{hash}". If we don't want to want to have the user there, we could urlbase64 it? Thoughts @jonatanklosko?

Reply:
perhaps we should change the dir name in both cases to `_2` suffix to not conflict with previous dirs created by Elixir? And if the dir is created by someone else then raising probably makes sense.

Reply:
> the reason we have a report is exactly because the permissions are confusing

I reproduced the problem in docker to make sure these changes work, but I don't have any objections to use multiple directories instead.

Reply:
@jonatanklosko I think the safest would be to have one directory per user, then we don't have to worry about anything else. It is unlikely different users are sharing the same compilation, right?